### PR TITLE
Added AME shielding deconstruction

### PR DESCRIPTION
**Pull request by Peptide90**
As title suggests. Players have been having issues placing AME in the wrong spot and can't break it down so made it so they can weld it back down to a part.

:cl:
- tweak: Antimatter Engine Shielding can now be deconstructed with a welder.

### CODE REVIEW

**Comment by Peptide90** [comment]
Not much clue what the error is to be honest so will look at this again over / after christmas break.

**Comment by Peptide90** [comment]
Wasn't keen on doing the same for the core as I don't know how easy we want it to be to take out power generation and 1 use of a welder seems too easy when previously they'd have to sabotage. Will leave that up to debate in future if we want to go full construction for AME.

File: Resources/Prototypes/Recipes/Construction/Graphs/utilities/ame_shielding.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+      edges:
+        - to: start
+          completed:
+            - !type:SpawnPrototype
+              prototype: AMEPart
+              amount: 1
+            - !type:DeleteEntity
+          steps:
+            - tool: Welding
+              doAfter: 3
```
Comment (metalgearsloth) [inline Resources/Prototypes/Recipes/Construction/Graphs/utilities/ame_shielding.yml:23]: camel case like the rest?

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

**Review by PaulRitter — COMMENTED** [review]
ideally i'd like there to only be a 20%-ish chance of actually being able to reclaim the ame part, but should be good for now.

that being said, i see no removal here of the original way of constructing the shielding, with a multitool. is there a reason to keep it?

**Comment by Peptide90** [comment]
> ideally i'd like there to only be a 20%-ish chance of actually being able to reclaim the ame part, but should be good for now.
> 
> that being said, i see no removal here of the original way of constructing the shielding, with a multitool. is there a reason to keep it?

This was meant as a stop gap for people miss placing the shielding and complaining they have to destroy it. This is for deconstruction only rather than construction and won't appear in the construction menu.

I mentioned moving the full construction to machine graphs but some people were opposed to that, but I can if need be or in future.

File: Resources/Prototypes/Recipes/Construction/utilities.yml [hunk 2]
```diff
@@ -46,6 +46,21 @@
...
+  startNode: start
+  targetNode: ameShielding
+  category: Utilities
+  description: "Shielding for the antimatter engine"
+  icon:
+    sprite: Structures/Power/Generation/ame.rsi
+    state: shield_0
+  objectType: Structure
+  placementMode: SnapgridCenter
+  canBuildInImpassable: false
```
Comment (Zumorica) [inline Resources/Prototypes/Recipes/Construction/utilities.yml:62]: Honestly, just remove this. The AME shielding should not be in the construction menu.

File: Resources/Prototypes/Recipes/Construction/Graphs/utilities/ame_shielding.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
+- type: constructionGraph
+  id: ameShielding
+  start: start
+  graph:
+    - node: start
+      edges:
+        - to: ameShielding
+          steps:
+            - prototype: AMEPart
+              doAfter: 2
```
Comment (Zumorica) [inline Resources/Prototypes/Recipes/Construction/Graphs/utilities/ame_shielding.yml:10]: If this is really only for deconstruction, then you don't need this.

**Review by Zumorica — CHANGES_REQUESTED** [review]

File: Resources/Prototypes/Recipes/Construction/Graphs/utilities/ame_shielding.yml (hunk 3 above)
Comment (Peptide90) [reply Resources/Prototypes/Recipes/Construction/Graphs/utilities/ame_shielding.yml]: Removed as requested. When I originally did this PR I was sure I was getting errors for not having complete graphs (no edge to node etc).

**Comment by Peptide90** [comment]
Happy for this to be canned if it's not desirable and we would rather go construction route or otherwise.

**Pull request merged**